downstream/dsurf/_Surface.py

```diff
@@ -38,23 +43,75 @@ def __init__(
             self._storage = storage
         self.algo = algo
 
-    def __iter__(self: "Surface") -> typing.Iterator[object]:
+    def __iter__(self: "Surface") -> typing.Iterator[typing.Optional[_Item]]:
         return iter(self._storage)
 
-    def __getitem__(self: "Surface", site: int) -> object:
+    def __getitem__(self: "Surface", site: int) -> typing.Optional[_Item]:
         return self._storage[site]
 
+    def __deepcopy__(self: "Surface", memo: dict) -> "Surface":
+        """An overloaded deepcopy to prevent a pickle error with cloning modules"""
+        return Surface(self.algo, deepcopy(self._storage, memo))
+
     @property
     def S(self: "Surface") -> int:
         return len(self._storage)
 
     def enumerate(
         self: "Surface",
-    ) -> typing.Iterable[typing.Tuple[int, object]]:
-        """Iterate over ingest times and values of retained data items."""
+    ) -> typing.Iterable[
+        typing.Tuple[typing.Optional[int], typing.Optional[_Item]]
+    ]:
+        """Iterate over ingest times and values of (possibly null) data items."""
```

Comment:
should specify that this is in order of layout on buffer surface (but with better words than layout on buffer surface).

Reply:
instead of possibly null --- , including sites not yet written to.